presto-spi/src/main/java/com/facebook/presto/spi/connector/ConnectorNodePartitioningProvider.java

```diff
@@ -52,4 +56,56 @@ BucketFunction getBucketFunction(
             ConnectorPartitioningHandle partitioningHandle,
             List<Type> partitionChannelTypes,
             int bucketCount);
+
+    class ConnectorBucketNodeMap
+    {
+        private final int bucketCount;
+        private final Optional<Map<Integer, Node>> bucketToNode;
+
+        public ConnectorBucketNodeMap(int bucketCount)
+        {
+            this(bucketCount, Optional.empty());
+        }
+
+        public ConnectorBucketNodeMap(Map<Integer, Node> bucketToNode)
+        {
+            this(
+                    bucketToNode.keySet().stream()
+                            .mapToInt(Integer::intValue)
+                            .max()
+                            .getAsInt() + 1,
```

Comment:
Are we sure that `getAsInt` will not throw?

Reply:
@nezihyigitbasi : Add `checkArgument` for empty map :)